packages/features/data-table/DataTableProvider.tsx

```diff
@@ -91,6 +96,12 @@ export function DataTableProvider({
   const [segmentId, setSegmentId] = useQueryState("segment", parseAsInteger.withDefault(-1));
   const [pageIndex, setPageIndex] = useQueryState("page", parseAsInteger.withDefault(0));
   const [pageSize, setPageSize] = useQueryState("size", parseAsInteger.withDefault(defaultPageSize));
+  const [searchTerm, setSearchTerm] = useQueryState("q", parseAsString.withDefault(""));
+
+  const setDebouncedSearchTerm = useMemo(
+    () => debounce((value: string | null) => setSearchTerm(value ? value.trim() : null), 500),
+    [setSearchTerm]
+  );
```

Comment:
Any reason we went with this debounce impl. over say our `useDebounce` hook?

Reply:
Yes! I'm trying not to call `setSearchTerm` function in the first place to avoid unnecessary re-renders.

With this approach, you get to call `setSearchTerm` function just a little bit, meaning much less re-rendering.

On the other hand, with useDebounce, you'd still call `setSearchTerm` a lot with every keystroke, meaning renredering DataTableProvider (although the debounced value will be stable, meaning it passes the same props to the children)

Reply:
Make sense! I see the difference now